src/build/commands.go

```diff
@@ -72,60 +72,60 @@ type EnvReplacableCommand interface {
 }
 
 // NewCommand make a new command according to the configuration given
-func NewCommand(cfg ConfigCommand) (cmd Command, err error) {
+func NewCommand(cfg ConfigCommand) (cmd Command) {
 	// TODO: use reflection?
 	switch cfg.name {
 	case "from":
-		cmd = &CommandFrom{cfg}
+		cmd = &CommandFrom{CommandBase{cfg}}
 	case "maintainer":
-		cmd = &CommandMaintainer{cfg}
+		cmd = &CommandMaintainer{CommandBase{cfg}}
 	case "run":
-		cmd = &CommandRun{cfg}
+		cmd = &CommandRun{CommandBase{cfg}}
 	case "attach":
-		cmd = &CommandAttach{cfg}
+		cmd = &CommandAttach{CommandBase{cfg}}
 	case "env":
-		cmd = &CommandEnv{cfg}
+		cmd = &CommandEnv{CommandBase{cfg}}
 	case "label":
-		cmd = &CommandLabel{cfg}
+		cmd = &CommandLabel{CommandBase{cfg}}
 	case "workdir":
-		cmd = &CommandWorkdir{cfg}
+		cmd = &CommandWorkdir{CommandBase{cfg}}
 	case "tag":
-		cmd = &CommandTag{cfg}
+		cmd = &CommandTag{CommandBase{cfg}}
 	case "push":
-		cmd = &CommandPush{cfg}
+		cmd = &CommandPush{CommandBase{cfg}}
 	case "copy":
-		cmd = &CommandCopy{cfg}
+		cmd = &CommandCopy{CommandBase{cfg}}
 	case "add":
-		cmd = &CommandAdd{cfg}
+		cmd = &CommandAdd{CommandBase{cfg}}
 	case "cmd":
-		cmd = &CommandCmd{cfg}
+		cmd = &CommandCmd{CommandBase{cfg}}
 	case "entrypoint":
-		cmd = &CommandEntrypoint{cfg}
+		cmd = &CommandEntrypoint{CommandBase{cfg}}
 	case "expose":
-		cmd = &CommandExpose{cfg}
+		cmd = &CommandExpose{CommandBase{cfg}}
 	case "volume":
-		cmd = &CommandVolume{cfg}
+		cmd = &CommandVolume{CommandBase{cfg}}
 	case "user":
-		cmd = &CommandUser{cfg}
+		cmd = &CommandUser{CommandBase{cfg}}
 	case "onbuild":
-		cmd = &CommandOnbuild{cfg}
+		cmd = &CommandOnbuild{CommandBase{cfg}}
 	case "mount":
-		cmd = &CommandMount{cfg}
+		cmd = &CommandMount{CommandBase{cfg}}
 	case "export":
-		cmd = &CommandExport{cfg}
+		cmd = &CommandExport{CommandBase{cfg}}
 	case "import":
-		cmd = &CommandImport{cfg}
+		cmd = &CommandImport{CommandBase{cfg}}
 	case "arg":
 		cmd = &CommandArg{CommandBase{cfg}}
 	default:
-		return nil, fmt.Errorf("Unknown command: %s", cfg.name)
+		panic(fmt.Sprintf("Unknown command: %s", cfg.name))
```

Comment:
It's generally preferable to not panic especially due to user error and not say `rocker` bug.